main/build.gradle

```diff
@@ -15,12 +15,36 @@
  */
 
 plugins {
-    id 'com.github.johnrengelman.shadow' version '5.2.0'
+    id 'com.github.johnrengelman.shadow' version '7.1.2'
     id 'application'
     id 'java'
+    id "com.diffplug.spotless" // Version provided in ../build.gradle.
     id 'maven-publish'
 }
 
+spotless {
```

Comment:
I don't believe you should need this, configuring once in the root `build.gradle` should be enough

Reply:
Uncertain. I think it wasn't running unless it was also defined in the sub directories. If I recall it was saying everything was fine, but not actually looking at the code under main.

I'm a little out of my depth and could use help/advice from a gradle java expert here.

Reply:
Ok I've run some experiments.

You're correct in that we don't need both, but it turns out the one we need is `./main/build.grade`.

What's needed in ./build.gradle is the spotless plugin version, but spotless (for whatever reason) won't look at .java files within ./main/ unless spotless is specifically configured within the main/build.gradle. This is different from how google-java-format works which is why it was confusing and took me a while to figure out (and how I ended up with redundant configuration here).

Made these changes in https://github.com/Trillium-Solutions/gtfs-validator/commit/4feba98edef19342ed3e9c376e01a98bc38c1559

Reply:
@barbeau are you happy with the changes here?

Reply:
@ed-g Do you think we could create another PR for the spotless task? We can discuss it there and then merge it if it's ready :)